Refresh the overlays + buttons when toggling the mobile preview

sobo-odoo opened the pull request.

robodoo commented:
This PR targets the un-managed branch odoo-dev/odoo:master-mysterious-egg, it needs to be retargeted before it can be merged.

robodoo commented:
This PR targets the un-managed branch odoo-dev/odoo:master-mysterious-egg, it needs to be retargeted before it can be merged.

loco-odoo commented on addons/html_builder/static/src/website_builder_action.js:
```diff
@@ -176,6 +178,7 @@ export class WebsiteBuilder extends Component {
 
     toggleMobile() {
         this.state.isMobile = !this.state.isMobile;
+        this.websitePreviewRef.el.classList.toggle("o_is_mobile");
```
nitpick: There should not be any problem but I think I would do `this.websitePreviewRef.el.classList.toggle("o_is_mobile", this.state.isMobile);` to be sure there is no inconsistency between `this.state.isMobile` and the presence of the `o_is_mobile` class

sobo-odoo commented:
@loco-odoo I made the [changes](https://github.com/odoo-dev/odoo/compare/96c995bd9c3f0bb3522ddf282d15e5d3cebe9e51..5a16a9a5316829d776e927ee5706a6160d7f896d) :slightly_smiling_face:
I also added a comment before toggling the class to explain why it is needed, and I removed the test for the overlay position, it was not really needed in the end and too overkill to make it work haha :slightly_smiling_face:
